src/js/plugin/sdk/pipeline/PipelineModel.js

```diff
@@ -0,0 +1,48 @@
+
+class PipelineModel {
+
+  constructor() {
+    this.pipelines = new Map();
+  };
+
+  clearPipeline() {
+    this.pipelines.clear();
+  };
+
+  applyFilter(operationName, filterParameters) {
+    const pipeline = this.getPipeline(operationName);
+
+    pipeline.forEach((filter) => {
+      filter(filterParameters);
+    });
+    return filterParameters;
+  };
+
+  register(operationName, callback) {
+
+    const pipeline = this.getPipeline(operationName);
+
+    if (pipeline.indexOf(callback) === -1) {
+      pipeline.push(callback);
+    }
+  };
+
+  deregister(operationName, callback) {
+    const pipeline = this.getPipeline(operationName);
+
+    var callbckPosition = pipeline.indexOf(callback);
+    if (callbckPosition !== -1) {
+      pipeline.splice(callbckPosition, 1);
+    }
+  };
+
+  getPipeline(pipelineName) {
+    if (!this.pipelines.has(pipelineName)) {
+      this.pipelines.set(pipelineName, []);
+    }
+    return this.pipelines.get(pipelineName);
+  };
+
+};
+
+export default new PipelineModel();
```

Comment:
Exporting an instance makes it unnecessary hard to test the `PipelineModel` and renders the store useless. Let's export the class here and use the store to create and provide the `PipelineModel` instance.